.github/workflows/ci.yml

```diff
@@ -57,52 +57,129 @@ jobs:
         fetch-depth: 0
 
     - name: Set up Python
-      if: matrix.python-version != 'winstore'
+      if: startswith(matrix.python-version, '3')
       uses: actions/setup-python@v4.6.0
       with:
         python-version: ${{ matrix.python-version }}
 
     - name: Install Windows Store Python
       if: matrix.python-version == 'winstore'
       uses: beeware/.github/.github/actions/install-win-store-python@main
-      with:
-        python-version: "3.11"
 
-    - name: Get packages
+    - name: Get Packages
       uses: actions/download-artifact@v3.0.2
       with:
         name: ${{ needs.package.outputs.artifact-name }}
         path: dist
 
-    - name: Install dev dependencies
+    - name: Install dev Dependencies
       run: |
-        # pip 23.1 has an issue with --user installs.
-        # See https://github.com/pypa/pip/issues/11982 for details
-        python -m pip install --upgrade "pip!=23.1"
+        python -m pip install --upgrade pip
         python -m pip install --upgrade setuptools
         # We don't actually want to install briefcase; we just
         # want the dev extras so we have a known version of tox.
         python -m pip install $(ls dist/briefcase-*.whl)[dev]
 
     - name: Test
+      env:
+        COVERAGE_FILE: ".coverage.${{ matrix.platform }}.${{ matrix.python-version }}"
       run: tox -e py --installpkg dist/briefcase-*.whl
 
-    - name: Store coverage data
+    - name: Store Coverage Data
+      if: success() || failure()
       uses: actions/upload-artifact@v3.1.2
       with:
         name: coverage-data
         path: ".coverage.*"
         if-no-files-found: ignore
 
-    - name: Report platform coverage
-      run: tox -e coverage
+  coverage-platform:
```

Comment:
If what we're generating is a "platform and python version" specific coverage report, and it will pass with 100% on every platform/python combination... is there any reason to *not* tag this to the end of the testing pass for a platform and python, rather than doing it as a separate step with it's own matrix fanout?

Reply:
In addition to wanting to provide a full analysis of coverage before failing, the only way to assess coverage across Python versions for a particular platform is to coalesce the coverage files at the end.

But this must be done using the OS for that platform. So, we end up with at least the 3 "Platform coverage" jobs alongside "Project coverage" after the tests.

At that point, if those 3 additional coverage jobs are necessary, the benefit of performing specific Python version coverage there becomes more palatable to 1) allow their failure to fail all of CI but not fail testing and 2) provide better consolidation of the coverage analysis in CI.

P.S.
> But [platform-specific coverage] must be done using the OS for that platform

This is necessary in order for the new conditional coverage rules to work. They operate on `sys.platform` to know when to require/ignore certain code blocks. We could consider spoofing this, though, to potentially allow coverage analysis of a platform while not on that platform. I can at least experiment with that.

Reply:
This turned out to be a fruitful experiment; I was able to incorporate python version-specific and platform-specific coverage reporting in-line with the existing job matrices.

Although, this has two larger status-quo changes:
- Unit tests no longer fast fail
  - This is to prevent coverage reporting from failing a job and thus the whole matrix.
  - I think ideally developers will have access to the full coverage information.
- The Verify App Build tests will not run now until unit tests and python version-specific coverage is passing.
  - Since failures for unit tests and python-specific coverage are represented by a single flag, either failure prevents app builds.
  - I haven't found an existing flag to avoid this....of course, we could always create one, I guess.

- #1262